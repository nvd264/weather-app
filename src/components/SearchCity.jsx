import React, { useState } from 'react';
import PropTypes from 'prop-types';

const SearchCity = ({ onSearch }) => {
  const [cityName, setCityName] = useState('');
  
  const handleSearchWeather = e => {
    if (e.key === 'Enter' && cityName) {
      onSearch(cityName);
    }
  };
  
  return (
    <div className="search-box">
      <input
        aria-label="search-input"
        onChange={(e) => setCityName(e.target.value)}
        onKeyPress={handleSearchWeather}
        type="text"
        className="form-control"
        id="cityName"
        placeholder="City name..."
        value={cityName}
      />
      {cityName && <span>&#9166; Enter</span>}
    </div>  
  )
};

SearchCity.propTypes = {
  onSearch: PropTypes.func.isRequired,
}

export default SearchCity;