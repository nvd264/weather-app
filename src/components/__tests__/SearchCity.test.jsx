import { render, screen, fireEvent } from '@testing-library/react';
import SearchCity from '../SearchCity';

describe('SearchCity Component', () => {
  
  const props = {
    onSearch: jest.fn(),
  };
  
  const setUp = () => {
    return render(<SearchCity {...props} />);
  };
  
  test('Should match the snapshot on init', () => {
    const { container } = setUp();
    expect(container).toMatchSnapshot();
  });
  
  test('Should show Enter suggestion when user input is entered', () => {
    const utils = setUp();
    const input = utils.getByLabelText('search-input');
    fireEvent.change(input, { target: { value: 'L' } });
    expect(screen.getByText(/Enter/)).toBeInTheDocument();
  });
  
  test('Should handle search when user press enter', () => {
    const utils = setUp();
    const input = utils.getByLabelText('search-input');

    fireEvent.change(input, { target: { value: 'London' } });
    fireEvent.keyPress(input, { key: "Enter", code: 13, charCode: 13 });
    
    expect(props.onSearch).toHaveBeenCalledWith('London');
  });
});


