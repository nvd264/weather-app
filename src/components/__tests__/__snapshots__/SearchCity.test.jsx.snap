// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCity Component Should match the snapshot on init 1`] = `
<div>
  <div
    class="search-box"
  >
    <input
      aria-label="search-input"
      class="form-control"
      id="cityName"
      placeholder="City name..."
      type="text"
      value=""
    />
    
  </div>
</div>
`;
